core/domain/classifier_services.py

```diff
@@ -100,8 +101,39 @@ def classify_string_classifier_rule(state, normalized_answer):
             training_examples.extend([
                 [doc, [str(answer_group_index)]]
                 for doc in classifier_rule_spec.inputs['training_data']])
-    if len(training_examples) > 0:
+
+    exploration = exp_services.get_exploration_by_id(exp_id)
+    state = exploration.states[state_name]
+    algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+        'TextInput']['classifier_id']
+    if state.classifier_model_id:
+        classifier = get_classifier_by_id(state.classifier_model_id)
+        sc_dict = classifier.cached_classifier_data
+        sc = classifier_registry.Registry.get_classifier_by_algorithm_id(
+            algorithm_id)
+        sc.from_dict(sc_dict)
```

Comment:
Seems reasonable to me, but note that some changes to ClassifierRegistry are implied in this, and those are generic to all algorithms. I'm OK with whatever implementation approach you decide to go with, given the deprecation of LDA, but just wanted to note that for reference,